### ISSUE TITLE
add validate_certs directive on unarchive module.

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29476]
From @ArKam on 2016-11-18T10:20:47Z

##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

unarchive
##### ANSIBLE VERSION

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```

##### OS / ENVIRONMENT

Ubuntu 16.04 LTS
##### SUMMARY

It would be nice to be able to allow a remote unarchive from self-signed or badly forged SSL HTTP Endpoint like it is possible with the validate_certs directive on the GET_URL module.
##### STEPS TO REPRODUCE

```

```
"N/A"

##### EXPECTED RESULTS

"N/A"

##### ACTUAL RESULTS

"N/A"

```

```

Copied from original issue: ansible/ansible-modules-core#5659

**Comment by ansibot** [comment]
From @ansibot on 2016-11-18T10:20:47Z



@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @dagwieers on 2016-11-18T10:20:47Z

As I am not a big fan to port every possible flag from get_url to unarchive and I fundamentally disagree that modules become more complex because it's more convenient to use a single action. So if others want to implement this and merge this, I won't object but I won't do it myself.

needs_contributor

**Comment by ansibot** [comment]
From @ArKam on 2016-11-18T10:20:47Z

That it's not a request for more complexity but rather to improve the way unarchive already handle SSL. That's a feature witch is mostely required in enterprise where you can't access outside repo but only internal trusted repo with SSL certificats that are provided by an internal CA. Ansible is mainly aiming to be use by enterprises.
It's a complete nonsens to have to use two steps (get_url + unarchive) only because of this issue.

**Comment by ansibot** [comment]
From @dagwieers on 2016-11-18T10:20:47Z

You are entitled to an opinion, as am I.

And it's not just because of this issue that I have this opinion, but rather as the maintainer of the unarchive module I received many requests for all possible URL fetching functionality and options that mimic what **get_url** (or **uri**) already implement. (authentication, proxy support, SSL functionality and options, client certificates, other schemes like ftp, sftp, smb, ...)

I am not interested in an unarchive module that is excellent in fetching files if we already have **get_url** with that sole purpose. And I will use my personal time in making unarchive better at unarchiving files.

**Comment by ansibot** [comment]
From @bcoca on 2016-11-18T10:20:47Z

@ArKam  please consider that @dagwieers is doing this as a volunteer, it is up to him what he wants to work on. 

That said, I mostly agree with him, the uri functionality is already fully supported in it's own module and copying all that functionality here is redundant. The increased complexity and maintenance is not worth the trouble to save 1 extra task from a playbook.

**Comment by Akasurde** [comment]
Closing this issue as this is implemented in version 2.2 (8c6af73)

**Comment by ArKam** [comment]
Hi @Akasurde I'm glad to read that! I would be curious to know what make people change their mind?

**Comment by Akasurde** [comment]
@ArKam I am not sure too. But happy to have this feature.
